src/pylife/stress/rainflow/recorders.py

```diff
@@ -139,3 +139,238 @@ def record_index(self, index_from, index_to):
+        if len(self._S_min) > 0 and type(self._S_min[0]) == pd.core.series.Series:
+            return self._get_for_every_node(self._is_zero_mean_stress_and_strain)
+        else:
+            return self._is_zero_mean_stress_and_strain
+
+    @property
+    def R(self):
+        """1-D numpy array containing the stress relation of the recorded hystereses,
+        which are usually computed as ``S_min / S_max``.
+        Only for hystereses resulting from Memory 3, the FKM nonlinear document defines ``R = -1``
+        (eq. 2.9-54). This is indicated by ``_is_zero_mean_stress_and_strain=True`̀ .
+        For these hystereses, this function returns -1 instead of ``S_min / S_max``, which may be different. """
+
+        return np.where(self.is_zero_mean_stress_and_strain, \
+                        -1, np.array(self._S_min) / np.array(self._S_max))
+
+    @property
+    def is_closed_hysteresis(self):
+        """1-D bool array indicating whether the row corresponds to a closed hysteresis or
+        was recorded as a memory 3 hysteresis, which counts only half the damage in the FKM nonlinear procedure."""
+
+        # if the assessment is performed for multiple points at once
+        if len(self._S_min) > 0 and type(self._S_min[0]) == pd.core.series.Series:
+            return self._get_for_every_node(self._is_closed_hysteresis)
+        else:
+            return self._is_closed_hysteresis
+
+    @property
+    def collective(self):
+        """The overall collective recorded as :class:`pandas.DataFrame`.
+
+        The load values are given in the columns ``loads_min``, and ``loads_max``
+        for consistency with other recoders.
+        Stress and strain values for the hystereses are given in the columns
+        ``S_min``, ``S_max``, and  ``epsilon_min``, ``epsilon_max``, respectively.
+        The column ``is_closed_hysteresis`` indicates whether the row corresponds
+        to a closed hysteresis or was recorded as a memory 3 hysteresis,
+        which counts only half the damage in the FKM nonlinear procedure.
+        The columns ``epsilon_min_LF`` and ``epsilon_max_LF`` describe the minimum
+        and maximum seen value of epsilon in the entire load history, up to the
+        current hysteresis. These values may be lower (min) or higher (max) than
+        the min/max values of the previously recorded hysteresis as they also
+        take into account parts of the stress-strain diagram curve that
+        are not part of hystereses.
+
+        The resulting DataFrame will have a MultiIndex with levels "hysteresis_index"
+        and "assessment_point_index", both counting from 0 upwards. The nodes of a mesh
+        are, thus, mapped to the index sequence 0,1,..., even if the
+        node_id starts, e.g., with 1.
+        """
+
+        # if the assessment is performed for multiple points at once
+        if len(self._S_min) > 0 and type(self._S_min[0]) == pd.core.series.Series:
```

Comment:
This also can likely be simplified. 

- [ ] Please file issue about refactoring after the merge.